### PR TITLE
merge DeepTauBase class into DeepTauId

### PR DESCRIPTION
**Pull request by yongbinfeng**
as the title suggested, PR to merge DeepTauBase into DeepTauId. 

New code developments are minor. Mostly to combine DeepTauBase.cc and DeepTauBase.h functions and classes into DeepTauId directly.

tested with `RecoTauTag/RecoTau/test/runDeepTauIDsOnMiniAOD.py` and the results on the TauIDs are consistent before and after change.

### CODE REVIEW

**Comment by mbluj** [comment]
@yongbinfeng, thank you for this development and sorry for belated review, but I was very busy last weeks. 
I went through the code and changes are fine for me, I am starting tests. 
@kandrosov, could also you check and sign off, please?

File: RecoTauTag/RecoTau/plugins/DeepTauId.cc [hunk 1]
```diff
@@ -1015,15 +1239,89 @@ class DeepTauId : public deep_tau::DeepTauBase {
     }
   }
 
+  void produce(edm::Event& event, const edm::EventSetup& es) {
```
Comment (mbluj) [inline RecoTauTag/RecoTau/plugins/DeepTauId.cc:1242]: please add `override` specifier (as suggested by `scram b code-checks`).

**Review by kandrosov — APPROVED** [review]
Thank you! The modification looks fine to me and it follows what we have discussed in the past. Although I didn't run any test and only did a visual inspection of the code.

File: RecoTauTag/RecoTau/plugins/DeepTauId.cc (hunk 1 above)
Comment (mbluj) [reply RecoTauTag/RecoTau/plugins/DeepTauId.cc:1242]: I will solve it myself as it small and unique issue.

**Comment by mbluj** [comment]
I have run the updated code and it works well. Therefore, I am going to merge this PR and then I will fix a small issue with "override" myself.

**Review by mbluj — APPROVED** [review]
Found issue is small and will be fixed after merge and before PR to official CMSSW.

**Comment by mbluj** [comment]
@yongbinfeng For further developments to enable deepTauID with SONIC a new branch called "CMSSW_13_2_X_tau-pog_deepTauSONIC" was created. The branch bases on 13_2_0_pre2 and contains changes introduced in this PR.

**Comment by yongbinfeng** [comment]
@mbluj @kandrosov thanks a lot for reviewing the PR! @mbluj thanks for fixing the remaining issue and preparing the 2nd branch!

i'm working on the 2nd PR now and should be ready by the end of this week.

**Comment by mbluj** [comment]
FYI, PR to official CMSSW: https://github.com/cms-sw/cmssw/pull/41841

**Pull request merged**